ADR/ADR037-implement-review-apps.md

```diff
@@ -0,0 +1,200 @@
+# ADD037: Implement review apps in a dedicated AWS account  
+
+Date: 2024-11-25
+
+## Status
+Proposed
+
+## Context
+
+### What are review apps?
+Review apps are short-lived deployments of applications, based on a branch other than the main branch. They are often created automatically in response to a new pull request, and are kept up to date with the branch automatically. They are usually destroyed when the pull request is closed.
+
+They exist to make it easy to create a live version of an application for review.
+
+In the GOV.UK Forms context, deploying a review copy of one app will inherently mean deploying a short-lived copy of another, because we operate a microservice architecture.
+
+### What are the limitations of review apps?
+
+Review apps do not do everything that a full copy of an environment does, because they are a quick-to-deploy subset of a total subsystem. Common limitations of review apps include not being able to talk to systems beyond the immediate bounds of the application[^external-systems], and available data being strictly limited to a known, pre-populated set of test data.   
+
+### Why do we want them
+
+We want to introduce review apps on PRs to support both technical and non-technical team members in gaining access to a working version of the code in question. For non-technical team members, introducing review apps will make it possible for them to make changes and test changes (usually content changes) for themselves, without the support of a developer. For engineer team members, review apps will make it easier for them to test a subset of the system in a production-ish environment without needing to disturb the fully fledged development environment.
+
+### What is in use elsewhere in GDS?
+
+Other teams in GDS have faced the same issues as us. One particularly good example is the GOV.UK Design System team, who have a mix of technical and non-technical members. They host prototypes and review versions of applications on [Heroku](https://www.heroku.com/) using Heroku's [Review Apps](https://devcenter.heroku.com/articles/github-integration-review-apps) feature. This is also used on GOV.UK Forms among our content and interaction designers for hosting prototypes.
+
+We investigated using Heroku Review Apps for our purposes, but we concluded that it was not a good choice. Heroku Review Apps is aimed at use cases involving a single application, but the applications we want to get review apps for will require deploying more than one application at a time.
+
+## Decision
+
+In this section we describe the architectural decisions we've made. Implementation details are out of scope.
```

Comment:
You don't need to state this as it is covered in https://gds-way.digital.cabinet-office.gov.uk/standards/architecture-decisions.html#what-an-adr-is-not

Instead, state the actual decision(s) as a longer version of the title, e.g.

> Implement Review Apps in a new, dedicated AWS account using AWS CodeBuild Managed Self-Hosted GitHub Action Runners to deploy new and updated pull requests to ECS. Each Review App is accessed via (something about URL structure) and routed to the correct container using Traefik.